text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1397 @@
+  efficiency, and data minimization require additional data processing capabilities such as data projection,
+  aggregation, and filtering.
+
+These improvements not only address the aforementioned needs but also answer the [open questions](https://github.com/open-telemetry/oteps/blob/main/text/0035-opentelemetry-protocol.md#open-questions)
+cited in OTEP 035 (i.e. cpu usage, memory pressure, compression optimization).
+
+**It is important to understand that this proposal is complementary to the existing protocol. The row-oriented version
+is still suitable for some scenarios. Telemetry sources that generate a small amount of telemetry data should continue
+to use it. On the other side of the spectrum, sources and collectors generating or aggregating a large amount of
+telemetry data will benefit from adopting this extension to optimize the resources involved in the transfer and
+processing of this data. This adoption can be done incrementally.**
+
+Before detailing the specifications of the OTLP Arrow protocol, the following two sections present: 1) a validation of
+the value of a columnar approach based on a set of benchmarks, 2) a discussion of the value of using Apache Arrow as a
+basis for columnar support in OTLP.
+
+### Validation
+
+A series of tests were conducted to compare compression ratios between OTLP and a columnar version of OTLP called OTLP
+Arrow. The key results are:
+
+* For univariate time series, OTLP Arrow is **2 to 3.5 better in terms of bandwidth reduction while having an
+  end-to-end speed (including conversion to/from OTLP) 1.2 to 1.5 times slower in phase 1**. In **phase 2** the conversion
+  OTLP to/from Arrow is gone and the end-to-end speed is **3.1 to 11.2 times faster by our estimates**.
+* For multivariate time series, OTLP Arrow is **4 times better in terms of bandwidth reduction while having an
+  end-to-end speed (including conversion to/from OTLP) similar to the univariate time series scenario phase 1**. Phase 2
+  has been yet estimated but similar results are expected.
+* For logs, OTLP Arrow is **2 to 3 times better in terms of bandwidth reduction while having an end-to-end speed
+  (including conversion to/from OTLP) 1.3 to 2 times slower in phase 1**. In **phase 2** the conversion
+  OTLP to/from Arrow is gone and the end-to-end speed is **2.3 to 4.86 times faster** by our estimates.
+* For traces, OTLP Arrow is **3 to 5 times better in terms of bandwidth reduction while having an end-to-end speed
+  (including conversion to/from OTLP) 1.5 to 2.1 times slower in phase 1**. In **phase 2** the conversion
+  OTLP to/from Arrow is gone and the end-to-end speed is **3.37 to 6.16 times faster** by our estimates.
+
+![Summary](img/0156_compression_ratio_summary.png)
+
+For both protocols, the baseline is the size of the uncompressed OTLP messages. The reduction factor is the ratio
+between this baseline and the compressed message size for each protocol. The compression algorithm used is ZSTD for
```

Comment:
The OTLP server doesn't support ZSTD at this time. https://github.com/open-telemetry/opentelemetry-collector/issues/7632 is open to track that need. How was ZSTD applied to OTLP here?